app/controllers/admin/routes.py

```diff
@@ -362,14 +367,49 @@ def addCourseFile():
     fileData = request.files['addCourseParticipants']
     filePath = os.path.join(app.config["files"]["base_path"], fileData.filename)
     fileData.save(filePath)
-    (session['errorFlag'], session['courseParticipantPreview'], session['previewCourseDisplayList']) = parseUploadedFile(filePath)
+    (session['cpPreview'], session['cpErrors']) = parseUploadedFile(filePath)
     os.remove(filePath)
-    return redirect(url_for("main.getAllCourseInstructors", showPreviewModal = True))
+    return redirect(url_for("admin.manageServiceLearningCourses"))
+
+@admin_bp.route('/manageServiceLearning', methods = ['GET', 'POST'])
+@admin_bp.route('/manageServiceLearning/<term>', methods = ['GET', 'POST'])
+def manageServiceLearningCourses(term=None):
+    """
+    The SLC management page for admins
+    """
+    if not g.current_user.isCeltsAdmin:
+        abort(403) 
+
+    if request.method =='POST' and "submitParticipant" in request.form:
+        saveCourseParticipantsToDatabase(session.pop('cpPreview', {}))
+        flash('Courses and participants saved successfully!', 'success')
+        return redirect(url_for('admin.manageServiceLearningCourses'))
+
+  
+    courseDict = getCourseDict()
```

Comment:
I think we could rename the `getCourseDict` to be more descriptive/accurate to what the function is doing.

Reply:
Yes, this was a terribly named function. It's not even close to what it is actually doing!